### PR TITLE
Don't memset SEvent v2

### PR DESCRIPTION
**Pull request by sfan5**
supersedes #15344

## To do

This PR is  Ready for Review.

## How to test

1. check gcc and clang CI for warnings
2. ensure that <s>Minetest</s> Luanti still generally works

### CODE REVIEW

File: irr/include/IEventReceiver.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ enum EEVENT_TYPE
 	to mouse or keyboard events. When a GUI element receives an event it will either
 	process it and return true, or pass the event to its parent. If an event is not absorbed
 	before it reaches the root element then it will then be passed to the user receiver. */
-	EET_GUI_EVENT = 0,
+	EET_GUI_EVENT = 1,
```
Comment (sfan5) [inline irr/include/IEventReceiver.h:21]: (this is extra safety if an unintialized event is accidentally used)

File: irr/include/IEventReceiver.h [hunk 2]
```diff
@@ -543,8 +543,9 @@ struct SEvent
 	};
 
 	SEvent() {
-		// would be left uninitialized in many places otherwise
-		MouseInput.Simulated = false;
+		EventType = static_cast<EEVENT_TYPE>(0);
+		// zero the biggest union member we have, which clears all others too
+		memset(&JoystickEvent, 0, sizeof(JoystickEvent));
```
Comment (swagtoy) [inline irr/include/IEventReceiver.h:548]: Just morbidly curious; is there a C++ way to deduce this? That would be clever.

Comment (sfan5) [reply irr/include/IEventReceiver.h:548]: with c++26 reflection: maybe
otherwise: no

Comment (swagtoy) [reply irr/include/IEventReceiver.h:548]: > c++26

Calm down son, we are still barely hanging onto C++17

File: irr/include/IEventReceiver.h [hunk 3]
```diff
@@ -543,8 +543,9 @@ struct SEvent
 	};
 
 	SEvent() {
-		// would be left uninitialized in many places otherwise
-		MouseInput.Simulated = false;
+		EventType = static_cast<EEVENT_TYPE>(0);
+		// zero the biggest union member we have, which clears all others too
```
Comment (SmallJoker) [inline irr/include/IEventReceiver.h:547]: Would a static assert be possible here?

e.g.
```C++
static_assert(sizeof(*this) - sizeof(EventType) == sizeof(JoystickEvent));
```

Comment (sfan5) [reply irr/include/IEventReceiver.h:547]: that wouldn't account for padding

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**